services/visual_recognition/v3.js

```diff
@@ -0,0 +1,364 @@
+/**
+ * Copyright 2013,2015, 2016 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+// Watson Visual Recognition functions supported by this node
+/*var FEATURE_RESPONSES = {
+  classifyImage: 'classifyImage',
+  detectFaces: 'detectFaces',
+  recognizeText: 'recognizeText',
+  createClassifier: 'createClassifier',
+  retrieveClassifiersList: 'retrieveClassifiersList',
+  retrieveClassifierDetails: 'retrieveClassifierDetails',
+  deleteClassifier: 'deleteClassifier'
+};*/
+
+module.exports = function (RED) {
+  var cfenv = require('cfenv');
+  var watson = require('watson-developer-cloud');
+
+  var imageType = require('image-type');
+  var url = require('url');
+  var temp = require('temp');
+  var fileType = require('file-type');
+  var fs = require('fs');
+  var async = require('async');
+
+  // temp is being used for file streaming to allow the file to arrive so it can be processed. 
+  temp.track();
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then the api key is stored in the variable s_apikey. 
+  //
+  // This separation between s_apikey and apikey is to allow 
+  // the end user to modify the key credentials when the service is not bound.
+  // Otherwise, once set apikey is never reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Taking this line out as codacy was complaining about it. 
+  // var services = cfenv.getAppEnv().services,
+
+  var apikey, s_apikey;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 's_apikey' is not in camel case.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2690779421/issues/source?bid=3330694&fileBranchId=3339165#l54)